src/main/java/seedu/revision/model/quiz/Statistics.java

```diff
@@ -15,68 +19,32 @@
     /*
      * The first character of the statistics must not be a whitespace,
      * otherwise " " (a blank string) becomes a valid input.
-     * 1 to 2 digits before and after the decimal point.
+     * 1 to more digits before and after the decimal point.
      */
     public static final String VALIDATION_REGEX = "(?i)\\d+\\/\\d+";
 
-    private int score;
-    private int total;
-    private double result = (double) score / total;
-    private int score1;
-    private int total1;
-    private int score2;
-    private int total2;
-    private int score3;
-    private int total3;
-
-    public Statistics(int score, int total, int score1, int total1, int score2, int total2, int score3, int total3) {
-        requireNonNull(score);
-        requireNonNull(total);
-        requireNonNull(score1);
-        requireNonNull(total1);
-        requireNonNull(score2);
-        requireNonNull(total2);
-        requireNonNull(score3);
-        requireNonNull(total3);
-        this.score = score;
-        this.total = total;
-        this.score1 = score1;
-        this.total1 = total1;
-        this.score2 = score2;
-        this.total2 = total2;
-        this.score3 = score3;
-        this.total3 = total3;
-    }
+    // Initialising a dictionary
+    private Dictionary<String, String> scores = new Hashtable<>();
 
-    public Statistics(String score) {
-        requireNonNull(score);
-        checkArgument(isValidStatistics(score), MESSAGE_CONSTRAINTS);
-        String[] splitStr = score.split(",", 4);
-        String[] splitScore = splitStr[0].split("/", 2);
-        String[] splitScore1 = splitStr[1].split("/", 2);
-        String[] splitScore2 = splitStr[2].split("/", 2);
-        String[] splitScore3 = splitStr[3].split("/", 2);
-        this.score = Integer.parseInt(splitScore[0]);
-        this.total = Integer.parseInt(splitScore[1]);
-        this.score1 = Integer.parseInt(splitScore1[0]);
-        this.total1 = Integer.parseInt(splitScore1[1]);
-        this.score2 = Integer.parseInt(splitScore2[0]);
-        this.total2 = Integer.parseInt(splitScore2[1]);
-        this.score3 = Integer.parseInt(splitScore3[0]);
-        this.total3 = Integer.parseInt(splitScore3[1]);
+    public Statistics(Dictionary scores) {
+        requireNonNull(scores);
```

Comment:
Did you assign score to any instance variable?

Reply:
changed